core/platform/email/mailgun_email_services.py

```diff
@@ -110,13 +110,26 @@ def send_bulk_mail(
     if not feconf.CAN_SEND_EMAILS:
         raise Exception('This app cannot send emails to users.')
 
-    data = {
-        'from': sender_email,
-        'to': recipient_emails,
-        'subject': subject,
-        'text': plaintext_body,
-        'html': html_body}
-
-    requests.post(
-        mailgun_domain_name, auth=('api', feconf.MAILGUN_API_KEY),
-        data=data)
+    # To send bulk emails we pass list of recipients in 'to' paarameter of
+    # post data. Maximum limit of recipients per request is 1000.
+    # For more detail check following link:
+    # https://documentation.mailgun.com/user_manual.html#batch-sending
+    recipient_emails = [
```

Comment:
Don't overwrite variables. Call this new one something like recipient_email_sets, and then do "for email_set in recipient_email_sets".